includes/class-migration.php

```diff
@@ -53,9 +53,20 @@ public static function get_version() {
 
 	/**
 	 * Locks the database migration process to prevent simultaneous migrations.
+	 *
+	 * @return bool|int True if the lock was successful, timestamp of existing lock otherwise.
 	 */
 	public static function lock() {
-		\update_option( 'activitypub_migration_lock', \time() );
+		global $wpdb;
+
+		// Try to lock.
+		$lock_result = $wpdb->query( $wpdb->prepare( "INSERT IGNORE INTO `$wpdb->options` ( `option_name`, `option_value`, `autoload` ) VALUES (%s, %s, 'no') /* LOCK */", 'activitypub_migration_lock', \time() ) ); // phpcs:ignore WordPress.DB
```

Comment:
what was wrong with `update_option`?

Reply:
It ignores any existing lock and updates it